src/oscar/core/loading.py

```diff
@@ -214,11 +214,12 @@ def _get_installed_apps_entry(app_name):
     This does depend on the order of INSTALLED_APPS and will break if
     e.g. 'dashboard.catalogue' comes before 'catalogue' in INSTALLED_APPS.
     """
-    for installed_app in settings.INSTALLED_APPS:
+    for entry in settings.INSTALLED_APPS:
+        app_config = AppConfig.create(entry)
         # match root-level apps ('catalogue') or apps with same name at end
         # ('shop.catalogue'), but don't match 'fancy_catalogue'
-        if installed_app == app_name or installed_app.endswith('.' + app_name):
-            return installed_app
+        if app_config.name == app_name or app_config.name.endswith('.' + app_name):
+            return app_config.name
     return None
 
 
```

Comment:
It should be possible to make the logic in `_find_installed_apps_entry` much more sane now. If we now require that the forked app uses the same label as the corresponding core app, then we should not have to mess around with sniffing through dotted paths to find a matching app? We should just be able to use [`apps.get_app_config()`](https://docs.djangoproject.com/en/2.0/ref/applications/#django.apps.apps.get_app_config) to fetch the appropriate app.